metrics/local.go

```diff
@@ -286,6 +286,11 @@ func (l *LocalFactory) newNamespace(name string) string {
 	if l.namespace == "" {
 		return name
 	}
+
+	if name == "" {
+		return l.namespace
+	}
```

Comment:
+1

NB: this kind of code is repeated several times in this library, it could use some refactoring.